src/components/atoms/Timeline/index.tsx

```diff
@@ -0,0 +1,244 @@
+      start: Math.max(end - DAY_SECS, 0),
+      end,
+    };
+  }
+
+  const defaultStart = Date.now();
+
+  return {
+    start: defaultStart,
+    end: defaultStart + DAY_SECS * EPOCH_SEC,
+  };
+};
+
+const Timeline: React.FC<Props> = memo(
+  function TimelinePresenter({ currentTime, range: _range, renderKnob, knobSize }) {
+    const range = useMemo(() => {
+      const range = getRange(_range);
+      if (process.env.NODE_ENV !== "production") {
+        if (range.start > range.end) {
+          throw new Error("Out of range error. `range.start` should be less than `range.end`");
+        }
+      }
+      return range;
+    }, [_range]);
+    const { start, end } = range;
+    const epochDiff = end - start;
+
+    // convert epoch diff to second.
+    const memoryCount = useMemo(
+      () => Math.trunc(epochDiff / EPOCH_SEC / MEMORY_INTERVAL),
+      [epochDiff],
+    );
+
+    const startDate = useMemo(() => new Date(start), [start]);
+    const hoursCount = Math.trunc(HOURS_SECS / MEMORY_INTERVAL);
+    // Convert memory count to pixel.
+    const currentPosition = useMemo(() => {
+      const diff = Math.min((currentTime - start) / EPOCH_SEC / MEMORY_INTERVAL, memoryCount);
+      const strongMemoryCount = diff / (hoursCount * STRONG_MEMORY_HOURS);
+      return Math.max(
+        diff * GAP_HORIZONTAL +
+          (diff - strongMemoryCount) * NORMAL_MEMORY_WIDTH +
+          strongMemoryCount * STRONG_MEMORY_WIDTH +
+          STRONG_MEMORY_WIDTH / 2,
+        0,
+      );
+    }, [currentTime, start, memoryCount, hoursCount]);
+
+    return (
+      <Container>
+        <MemoryBox>
+          <MemoryList start={startDate} memoryCount={memoryCount} hoursCount={hoursCount} />
+          <Icon
+            style={{
+              left: currentPosition + PADDING_HORIZONTAL - knobSize / 2,
+            }}>
+            {renderKnob()}
+          </Icon>
+        </MemoryBox>
+      </Container>
+    );
+  },
+  (prev, next) => prev.currentTime === next.currentTime,
+);
+
+const Container = styled.div`
+  background: ${({ theme }) => theme.main.lighterBg};
+  width: 100%;
+  display: flex;
+  align-items: center;
+  padding: 5px 12px;
+  box-sizing: border-box;
+`;
+
+const MemoryBox = styled.div`
+  border: 1px solid #b5b5b5;
+  border-radius: 5px;
+  box-sizing: border-box;
+  position: relative;
+  overflow-x: auto;
+  overflow-y: hidden;
+  width: 100%;
+  ::-webkit-scrollbar {
+    height: 5px;
+  }
+  ::-webkit-scrollbar-track {
+    background-color: transparent;
+  }
+  ::-webkit-scrollbar-thumb {
+    border-radius: 5px;
+    background-color: #aaa;
+  }
+`;
+
+const Icon = styled.div`
+  position: absolute;
+  top: 2px;
+`;
+
+const MONTH_LABEL_LIST = [
+  "Jan",
+  "Feb",
+  "Mar",
+  "Apr",
+  "May",
+  "Jun",
+  "Jul",
+  "Aug",
+  "Sep",
+  "Oct",
+  "Nov",
+  "Dec",
+];
+
+type MemoryListProps = {
+  start: Date;
+  memoryCount: number;
+  hoursCount: number;
+};
+
+const MemoryList: FC<MemoryListProps> = memo(
```

Comment:
Above you use `React.FC` and here you do `FC`. Can you use one or the other for consistency? 
Personally I feel like `React.FC` is clearer, but if you feel FC is enough please let me know.

Reply:
Also, could you move this component to the bottom of the page. Currently it is in the middle of the styled components and it would be nicer to keep things organized.

Reply:
> Above you use React.FC and here you do FC. Can you use one or the other for consistency?
Personally I feel like React.FC is clearer, but if you feel FC is enough please let me know.

this is fixed by [type: use Recat.FC](https://github.com/reearth/reearth-web/pull/256/commits/84f58a432988f94866fb1aafec4e5346fe2e771d)

> Also, could you move this component to the bottom of the page. Currently it is in the middle of the styled components and it would be nicer to keep things organized.

this is fixed by [refactor: move style to bottom](https://github.com/reearth/reearth-web/pull/256/commits/88054be9ddff7a84eee271d31dcf45c14b16c1ab)